src/dswx_sar/filter_SAR.py

```diff
@@ -122,11 +125,210 @@ def lee_enhanced_filter(img, win_size=3, k=K_DEFAULT, cu=CU_DEFAULT,
+    win_size = kwargs.get('win_size', -1)
 
     # we process the entire img as float64 to avoid type overflow error
     img = np.float64(img)
     w_t, mean, _ = weightingarr(img, win_size, k, cu, cmax)
     filter_im = (mean * w_t) + (img * (1 - w_t))
 
     return filter_im
+
+
+def fill_nan_value(data):
+    mask = np.isnan(data) | np.isinf(data) | np.isinf(-data)
+    data[mask] = -15
+    return data
+
+
+def find_largest_nan_polygon(data):
+    """
+    Identifies contiguous NaN regions and finds the largest one.
+
+    :param data: 2D numpy array with NaNs.
+    :return: The label of the largest NaN region and an array where each
+             contiguous NaN region is labeled with a unique integer.
+    """
+    # Identify NaNs and invert, because ndimage.label labels non-zero regions
+    nan_mask = np.isnan(data)
+    labeled_regions, num_features = ndimage.label(~nan_mask)
+
+    # Count the occurrences of each label (excluding the background, which is 0)
+    label_sizes = np.bincount(labeled_regions.ravel())[1:]  # Exclude label 0
+
+    # Find the label of the largest NaN region
+    largest_label = np.argmax(label_sizes) + 1  # +1 because we excluded label 0 earlier
+
+    return largest_label, labeled_regions
+
+
+def interpolate_except_largest_nan_region(data):
+    """
+    Interpolates to fill NaN values for areas not covered by the largest contiguous NaN region.
+
+    :param data: 2D numpy array with NaNs.
+    :return: 2D numpy array with NaNs filled, except for the largest NaN region.
+    """
+    nan_mask = np.isnan(data)
+    largest_label, labeled_regions = find_largest_nan_polygon(data)
+
+    # Mask for the largest NaN region, which we'll exclude from filling
+    largest_nan_mask = (labeled_regions == largest_label)
+
+    # Find indices of NaNs that are not in the largest NaN region
+    fill_indices = np.where(nan_mask & ~largest_nan_mask)
+
+    # Coordinates of points for interpolation (excluding largest NaN region and non-NaN)
+    points = np.column_stack((np.where(~nan_mask & ~largest_nan_mask)))
+    values = data[~nan_mask & ~largest_nan_mask]
+
+    # Coordinates where we want to interpolate (excluding largest NaN region)
+    xi = np.column_stack(fill_indices)
+
+    if xi.size > 0 and values.size > 0:  # Ensure there are points to interpolate
+        # Perform interpolation
+        interpolated_values = griddata(points, values, xi, method='nearest')
+
+        # Insert interpolated values back into the original data array
+        data[fill_indices] = interpolated_values
+    else:
+        data = -30
+    return data
+
+
+def anisotropic_diffusion(img, **kwargs):
+    """
+    Anisotropic Diffusion
+
+    Parameters
+    ----------
+    img: numpy.ndarray
+        2 dimensional array which has real intensity values
+    weight: float
+        window size to apply filter
+
+    Returns
+    -------
+    filter_im: numpy.ndarray
+        filtered intensity image.
+    """
```

Comment:
```suggestion
    NOTE: [link to the brief description about the filter]
    """
```
And same for `guided_filter` and `tv_bregman` as well, in case the document is available online.